### PR TITLE
chore: improve sql performance by relying on Hibernate optimizations

### PR DESCRIPTION
**Pull request by bjagg**

### CODE REVIEW

File: fbms-core/src/main/java/org/apereo/portal/fbms/data/SubmissionRepository.java [hunk 1]
```diff
@@ -26,16 +26,8 @@
...
-            "WHERE s.id.username = :username " +
-            "AND s.id.fname = :fname " +
-            "ORDER BY s.id.timestamp DESC")
-    List<SubmissionEntity> findByUsernameAndFname(@Param("username") String username, @Param("fname") String fname);
+    //List<SubmissionEntity> findByUsernameAndFnameOrderByTimestampDesc(String username, String fname);
 
-    @Query("SELECT s FROM SubmissionEntity s " +
-            "WHERE s.id.timestamp = (SELECT MAX(b.id.timestamp) from SubmissionEntity b WHERE b.id.username = :username AND b.id.fname = :fname) " +
-            "AND s.id.username = :username " +
-            "AND s.id.fname = :fname")
```
Comment (ChristianMurphy) [inline fbms-core/src/main/java/org/apereo/portal/fbms/data/SubmissionRepository.java:38]: 🤷‍♂️ I'll defer to others on whether to go the hibernate magic query route.
The SQL could be optimized without Hibernate magic with:

```sql
SELECT s
FROM SubmissionEntity s
WHERE s.id.username = :username
AND s.id.fname = :fname
ORDER BY s.id.timestamp DESC
LIMIT 1
```

to eliminate the inner query, which is likely the bottleneck here.

File: fbms-core/src/main/java/org/apereo/portal/fbms/data/FormRepository.java [hunk 2]
```diff
@@ -24,11 +24,8 @@
 
 public interface FormRepository extends CrudRepository<FormEntity,FormIdentifier> {
 
-    default boolean existsByFname(String fname) {
-        return findMostRecentByFname(fname) != null;
-    }
+    boolean existsByFname(String fname);
 
-    @Query("SELECT f FROM FormEntity f WHERE f.id.version = (SELECT MAX(b.id.version) from FormEntity b WHERE b.id.fname = :fname) AND f.id.fname = :fname")
```
Comment (ChristianMurphy) [inline fbms-core/src/main/java/org/apereo/portal/fbms/data/FormRepository.java:31]: 🤷‍♂️ I'll defer to others on whether to go the hibernate magic query route.
The SQL could be optimized without Hibernate magic with:

```sql
SELECT f
FROM FormEntity f
WHERE f.id.fname = :fname
ORDER BY f.id.version DESC
LIMIT 1
```

to eliminate the inner query, which is likely the bottleneck here.

**Review by ChristianMurphy — APPROVED** [review]
⚠️ I have not spun up an instance to test this yet ⚠️ 

From a code perspective this looks good, thanks @bjagg!
I go back and forth on whether or not hibernate automatically generated queries are a good idea or not.
And offered some alternatives below if there is a preference to keep the SQL in place.

File: fbms-core/src/main/java/org/apereo/portal/fbms/data/SubmissionRepository.java (hunk 1 above)
Comment (bjagg) [reply fbms-core/src/main/java/org/apereo/portal/fbms/data/SubmissionRepository.java:38]: There was some back and forth in a few discussion threads about some DBs  *NOT* honoring `LIMIT 1`. By letting Spring and Hibernate optimize based on DB dialect, we can avoid this concern.

Appreciate the HQL @ChristianMurphy !

**Comment by bjagg** [comment]
To facilitate testing, I have captured an FBMS test setup in a branch: https://github.com/bjagg/uPortal-start/tree/test/fbms-test

**Review by loulou2u — APPROVED** [review]
Tested FBMS and it works! 👍

**Pull request merged**